packages/flutter/test/widgets/slivers_keepalive_test.dart

```diff
@@ -0,0 +1,678 @@
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.children != widget.children) {
+      children = widget.children;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Directionality(
+      textDirection: TextDirection.ltr,
+      child: Center(
+        child: Container(
+          height: 100,
+          child: CustomScrollView(
+            cacheExtent: 0,
+            slivers: <Widget>[
+              SliverFillViewport(
+                viewportFraction: widget.viewportFraction,
+                delegate: SliverChildListDelegate(children),
+              )
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+}
+
+class SwitchingSliverListTest extends StatefulWidget {
+  SwitchingSliverListTest({
+    this.children,
+    this.viewportFraction = 1.0,
+    Key key}) : super(key: key);
+
+  final List<Widget> children;
+  final double viewportFraction;
+
+  @override
+  _SwitchingSliverListTest createState() => _SwitchingSliverListTest();
+
+}
+
+class _SwitchingSliverListTest extends State<SwitchingSliverListTest> {
+  List<Widget> children;
+
+  @override
+  void initState() {
+    super.initState();
+    children = widget.children;
+  }
+
+  @override
+  void didUpdateWidget(SwitchingSliverListTest oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.children != widget.children) {
+      children = widget.children;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Directionality(
+      textDirection: TextDirection.ltr,
+      child: Center(
+        child: Container(
+          height: 100,
+          child: CustomScrollView(
+            cacheExtent: 0,
+            slivers: <Widget>[
+              SliverList(
+                delegate: SliverChildListDelegate(children),
+              )
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+}
+
+class WidgetTest1 extends StatefulWidget {
+  WidgetTest1({
+    this.text,
+    this.keepAlive = false,
+    Key key}) : super(key: key);
+
+  final String text;
+  final bool keepAlive;
+
+  @override
+  String toString({DiagnosticLevel minLevel = DiagnosticLevel.debug}) {
+    return 'WidgetTest1 $text';
+  }
+
+  @override
+  _WidgetTest1 createState() => _WidgetTest1();
+
+}
+
+class _WidgetTest1 extends State<WidgetTest1> with AutomaticKeepAliveClientMixin{
+  bool hasBeenDisposed = false;
+
+  @override
+  Widget build(BuildContext context) {
+    super.build(context);
+    return Text(widget.text);
+  }
+
+  @override
+  void dispose() {
+    hasBeenDisposed = true;
+    super.dispose();
+  }
+
+  @override
+  bool get wantKeepAlive => widget.keepAlive;
+
+}
+
+class WidgetTest2 extends StatefulWidget {
+  WidgetTest2({
+    this.text,
+    this.keepAlive = false,
+    Key key}) : super(key: key);
+
+  final String text;
+  final bool keepAlive;
+
+  @override
+  String toString({DiagnosticLevel minLevel = DiagnosticLevel.debug}) {
```

Comment:
nit: add spaces after `{` and before `}`.

Reply:
I forgot to remove this function. removed